### ISSUE TITLE
Use same proxy everytimebut randomize others

### ISSUE THREAD

**Issue by sinnlosername** [rofl0r/proxychains-ng#197]
Hei.

Is it possible to use the same proxy as first proxy everytime, but randomize other proxys at every connection? I want to use tor as first proxy for everything, but I want to randomize my output ip.

**Comment by rofl0r** [comment]
yes, it is possible. you just need to code it.

**Comment by sinnlosername** [comment]
This answer is really helpful. C is a very ugly language, so could you tell me where I need to put code? I think, you, as developer, know where random proxy's are selected. D:

**Comment by karmamaster** [comment]
I just wondered if YOU are not a developer so how can YOU use this? Do it by your self first and ask when you have tried all things you have found.

**Comment by rofl0r** [comment]
C is not ugly, it's beautiful and elegant in its simplicity. what is ugly is if people that don't know better litter it with `#ifdef`'s on their quest of doing non-portable things... anyway, look at the function `select_proxy()` in core.c, which is responsible for selecting the proxy, as its name suggests. you may also want to compile proxychains with `-DDEBUG` in CFLAGS to get more debug output.

**Comment by sinnlosername** [comment]
@karmamaster I already looked through the code, before I wrote this ticket. Yes, I'm a developer, but I never devleoped something in c, but I think I'll be able to do some array operations. If I would know where those proxies are selected, I could just set the first proxy to the proxy I want.

@rofl0r No. I'm good in java & golang. I'm also able to develope small things in php or bash. But even with developing skills in those languages, I'm unable to understand most code of proxychains. So the language is very different or just very hard to understand. ^^
